wasm/main.go

```diff
@@ -1,69 +1,674 @@
-package main
+// GOOS=js GOARCH=wasm go build -o  ../assets/hive.wasm
 
-//go:generate GOOS=js GOARCH=wasm go build -o  ../assets/hive.wasm
+package main
 
 import (
+	"bufio"
 	"bytes"
 	"encoding/json"
+	"fmt"
+	"github.com/hako/durafmt"
+	logger "github.com/ipfs/go-log/v2"
 	"io/ioutil"
 	"net/http"
+	"reflect"
+	"strings"
 	"syscall/js"
-
-	logger "github.com/ipfs/go-log/v2"
+	"time"
 )
 
-var log = logger.Logger("sock/server")
+var log = logger.Logger("events")
+
 const (
+	EVENTS  = "http://localhost:4343/v3/events"
 	GATEWAY = "http://localhost:4343/v3/execute"
 )
 
-func ID() js.Func {
+var StartTime int64
+
+func Events() js.Func {
+
+	jsonfunc := js.FuncOf(func(this js.Value, args []js.Value) interface{} {
+
+		go func() {
+			log.Debug("Events Called")
+			resp, err := http.Post(EVENTS, "application/json", nil)
+			if err != nil {
+				log.Error(err.Error())
+				return
+			}
+			defer resp.Body.Close()
+			reader := bufio.NewReader(resp.Body)
+			for {
+				var eventsDataString string
+				line, isPrefix, err := reader.ReadLine()
+				log.Debugf("This is line: %s", string(line))
+				if string(line) == "" {
+					log.Debug("Empty Response at reader.ReadLine")
+					return
+				}
+				if err != nil {
+					log.Debugf("Error in reading data string: %s", err.Error())
+					continue
+				}
+				eventsDataString = string(line)
+				for isPrefix {
+					line, isPrefix, err = reader.ReadLine()
+					if err != nil {
+						log.Debugf("Error in reading prefixed data string: %s", err.Error())
+						continue
+					}
+					eventsDataString += string(line)
+				}
+				log.Debugf("This is the Events Data String: %+v", eventsDataString)
+				var event Event
+				err = json.Unmarshal([]byte(eventsDataString), &event)
+				if err != nil {
+					log.Error("Error in Unmarshalling eventsDataString:", err.Error())
+					return
+				}
+				var out Out
+				log.Debugf("This is event: %s", event.Result.Topic)
+				err = json.Unmarshal([]byte(event.Result.Val), &out)
+				if err != nil {
+					log.Error("Error in Unmarshalling Out in : ", event.Result.Val, err.Error())
+					return
+				}
+				val, err := json.Marshal(out.Data)
+				if err != nil {
+					log.Error("Error encountered in Marshalling: ", err.Error())
+					return
+				}
+				if (event.Result.Topic == "Status") || (event.Result.Topic == "Balance") || (event.Result.Topic == "BalanceCycle") || (event.Result.Topic == "Peers") || (event.Result.Topic == "Settings") || (event.Result.Topic == "Settlement") {
```

Comment:
Is this 'if' case required?

If the topic is not in the switch case, we will hit default case which is a no-op.

Ideally you just need to make sure len(event.Result.Topic) > 0 here.